src/Microsoft.AspNet.OData.Shared/EnableQueryAttribute.cs

```diff
@@ -0,0 +1,783 @@
+        /// </item>
+        /// <item>
+        /// <term>DateTime related:</term>
+        /// <description>year, month, day, hour, minute, second, fractionalseconds, date, time
+        /// e.g. ~/Employees?$filter=year(BirthDate) eq 1971</description>
+        /// </item>
+        /// <item>
+        /// <term>Math related:</term>
+        /// <description>round, floor, ceiling</description>
+        /// </item>
+        /// <item>
+        /// <term>Type related:</term>
+        /// <description>isof, cast</description>
+        /// </item>
+        /// <item>
+        /// <term>Collection related:</term>
+        /// <description>any, all</description>
+        /// </item>
+        /// </list>
+        /// </summary>
+        public AllowedFunctions AllowedFunctions
+        {
+            get
+            {
+                return _validationSettings.AllowedFunctions;
+            }
+            set
+            {
+                _validationSettings.AllowedFunctions = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value that represents a list of allowed arithmetic operators including 'add', 'sub', 'mul',
+        /// 'div', 'mod'.
+        /// </summary>
+        public AllowedArithmeticOperators AllowedArithmeticOperators
+        {
+            get
+            {
+                return _validationSettings.AllowedArithmeticOperators;
+            }
+            set
+            {
+                _validationSettings.AllowedArithmeticOperators = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value that represents a list of allowed logical Operators such as 'eq', 'ne', 'gt', 'ge',
+        /// 'lt', 'le', 'and', 'or', 'not'.
+        /// </summary>
+        public AllowedLogicalOperators AllowedLogicalOperators
+        {
+            get
+            {
+                return _validationSettings.AllowedLogicalOperators;
+            }
+            set
+            {
+                _validationSettings.AllowedLogicalOperators = value;
+            }
+        }
+
+        /// <summary>
+        /// <para>Gets or sets a string with comma separated list of property names. The queryable result can only be
+        /// ordered by those properties defined in this list.</para>
+        ///
+        /// <para>Note, by default this string is null, which means it can be ordered by any property.</para>
+        ///
+        /// <para>For example, setting this value to null or empty string means that we allow ordering the queryable
+        /// result by any properties. Setting this value to "Name" means we only allow queryable result to be ordered
+        /// by Name property.</para>
+        /// </summary>
+        public string AllowedOrderByProperties
+        {
+            get
+            {
+                return _allowedOrderByProperties;
+            }
+            set
+            {
+                _allowedOrderByProperties = value;
```

Comment:
Small security consideration: should we limit the input to alphanumeric characters? I don't think people in general would try to put random symbols, but just a thought.

Reply:
This is existing code refactored into Shared so it's look new but it's now. That does not answer the question but it is a reason to leave it as-is.